### ISSUE TITLE
DropDownMenu cuts off, if it contains too many items

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1388]
1.0.0-beta6-dev455

From [this comment](https://github.com/JetBrains/compose-jb/issues/1329#issuecomment-966652458)

```
import androidx.compose.foundation.layout.Box
import androidx.compose.material.Button
import androidx.compose.material.DropdownMenu
import androidx.compose.material.DropdownMenuItem
import androidx.compose.material.Text
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication {
    var expanded by remember { mutableStateOf(false) }
    Box {
        Button(
            onClick = { expanded = true }
        ) {
            Text("Open the dropdown")
        }
        DropdownMenu(
            expanded = expanded,
            onDismissRequest = { expanded = false }
        ) {
            DropdownMenuItem(onClick = {}) { Text("First item") }
            DropdownMenuItem(onClick = {}) { Text("Second item") }
            DropdownMenuItem(onClick = {}) { Text("Third item") }
            DropdownMenuItem(onClick = {}) { Text("Fourth item") }
            DropdownMenuItem(onClick = {}) { Text("Fifth item") }
            DropdownMenuItem(onClick = {}) { Text("Sixth item") }
            DropdownMenuItem(onClick = {}) { Text("Seventh item") }
            DropdownMenuItem(onClick = {}) { Text("Eighth item") }
        }
    }
}

```

1. Open example
2. Press the Button
3. Shrink the height of the window

Result:
![image](https://user-images.githubusercontent.com/5963351/141434991-bed2b29a-a6cd-4cf6-8294-b863caed0516.png)

I expect this is because we place Popup above anchor point, if it doesn't fit below it.

`DropdownMenu` has `Modifier.verticalScroll` by default. Somewhere we don't calculate the size of the popup in a right way.

**Comment by Thijsiez** [comment]
I tested with `v1.0.0-beta6-dev462` earlier and nothing seems to have changed. Probably because [Jetbrains/androidx#79](https://github.com/JetBrains/androidx/pull/79) has not been accepted yet

**Comment by akurasov** [comment]
Yes, it is not merged yet

**Comment by akurasov** [comment]
Fixed in 0.0.0-master-dev473, so will be in next 1.0.0-beta6-*

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6216">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
